lib/determinator/tracking/rack/middleware.rb

```diff
@@ -16,7 +16,24 @@ def call(env)
           error = true
           raise
         ensure
-          Determinator::Tracking.finish!(status: status, error: !!error)
+          Determinator::Tracking.finish!(
+            status: status,
+            error: !!error,
+            endpoint: extract_endpoint(env)
+          )
+        end
+
+        private
+
+        def extract_endpoint(env)
+          parts = if params = env['action_dispatch.request.path_parameters']
+            [[params[:controller], params[:action]].join('#')]
+          else
+            [env['REQUEST_METHOD'], env['PATH_INFO'] || env['REQUEST_URI']]
+          end
```

Comment:
I feel like treating Rails apps differently (vs every other Rack app) is kinda weird - would it not be better to be consistent with using the Rack request environment for tracking?

Reply:
That's a fair point, but the goal here is to extract a meaningful identifier; and considering the vast majority of the usages are going to be in Rails app, it feels worthwhile using the the information if available.